cmake/module/GenerateSymlinks.cmake

```diff
@@ -0,0 +1,88 @@
+# Function: create_symlink_for_target
+# -----------------------------------
+# Creates a symlink for a given CMake target so it can be built in
+# CMAKE_RUNTIME_OUTPUT_DIRECTORY but remains accessible in its default subdirectory
+# location.
+#
+# Parameters:
+# - target_name: The name of the CMake target (must be defined via `add_executable`
+#   or `add_library`).
+#
+# Example:
+#   create_symlink_for_target(my_executable)
+#   - If my_executable previously resided in `CMAKE_CURRENT_BINARY_DIR/my_executable`,
+#     but is now in `CMAKE_RUNTIME_OUTPUT_DIRECTORY/my_executable`, a symlink will be created in
+#     the original location pointing to the new location.
+#
+# This function is intended to be called automatically for all targets in a project.
+
+function(create_symlink_for_target target_name)
+    get_target_property(target_type ${target_name} TYPE)
+    get_target_property(target_old_path ${target_name} BINARY_DIR)
+    if(target_type STREQUAL "EXECUTABLE")
+        set(target_new_path "${CMAKE_RUNTIME_OUTPUT_DIRECTORY}")
+        set(target_output_name "${CMAKE_EXECUTABLE_PREFIX}${target_name}${CMAKE_EXECUTABLE_SUFFIX}")
+    else()
+        return()
+    endif()
+
+    add_custom_target(${target_name}_create_symlink
```

Comment:
I wonder if this should be an `add_custom_command()` with a [`POST_BUILD` event](https://cmake.org/cmake/help/latest/command/add_custom_command.html#build-events) instead? Presumably this version (I haven't checked) creates the symlink even if the link fails, whereas I would expect a `POST_BUILD`command to not have that problem.

Reply:
What problem do you see this solving? Creating a symlink pointing at a target that does not exist should be fine. It will still point to the right place.